modules/video/test/ocl/test_optflowpyrlk.cpp

```diff
@@ -121,13 +124,24 @@ OCL_TEST_P(PyrLKOpticalFlow, Mat)
             float errdiff = 0.0f;
 
             if (!eq || errdiff > 1e-1)
+            {
                 ++mistmatch;
+                continue;
+            }
+
+            eq = std::abs(cpuErr[i] - err[i]) < 0.01;
+            if(!eq &&
+               winSize.width >= 16 && winSize.height >=16 &&
+               winSize.width <=24 && winSize.height <=24)
```

Comment:
1)
I mean there is no reason to call some implementation (for example, OpenCL kernel) with unsupported parameters.

You state this:
> In this kernel, the input winSize bigger than 24x24 will be referred as 24x24, the input winSize smaller than 16x16 will be referred as 16x16.

if this is true, then appropriate parameters check should be added.

2)
All OpenCV tests should not rely on concrete implementation.
So this strange condition should be eliminated, because its justification is related to concrete OpenCL implementation and it will not be accepted.